src/engine/pieces/knight.ts

```diff
@@ -1,13 +1,32 @@
-import Piece from './piece';
+import Piece, {MoveDiagnostic, PieceType} from './piece';
 import Player from '../player';
 import Board from '../board';
 
 export default class Knight extends Piece {
     public constructor(player: Player) {
-        super(player);
+        super(player, PieceType.KNIGHT);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let validMoves = new Array(0);
+        let currentSquare = board.findPiece(this);
+
+        let directions = [
+            {"rowChange": 1, "colChange": 2},
+            {"rowChange": 2, "colChange": 1},
+            {"rowChange": -1, "colChange": 2},
+            {"rowChange": 2, "colChange": -1},
+            {"rowChange": 1, "colChange": -2},
+            {"rowChange": -2, "colChange": 1},
+            {"rowChange": -1, "colChange": -2},
+            {"rowChange": -2, "colChange": -1},
+        ];
+
+        for (let direction of directions){
```

Comment:
nitpick but Instead of a for loop here we could use `forEach `

e.g. 
`directions.forEach(direction => {`